vendor/github.com/RangelReale/osin/access.go

```diff
@@ -221,6 +227,34 @@ func (s *Server) handleAuthorizationCodeRequest(w *Response, r *http.Request) *A
 		return nil
 	}
 
+	// Verify PKCE, if present in the authorization data
+	if len(ret.AuthorizeData.CodeChallenge) > 0 {
+		// https://tools.ietf.org/html/rfc7636#section-4.1
+		if matched := pkceMatcher.MatchString(ret.CodeVerifier); !matched {
+			w.SetError(E_INVALID_REQUEST, "code_verifier invalid (rfc7636)")
+			w.InternalError = errors.New("invalid format")
```

Comment:
The internal errors show up in the server log.  It would be nice if the error was `code_verifier invalid format` so you don't have to figure out _what_ is invalid (I ended up putting debug print statements when working on the browser CLI flow in places where it wasn't possible to set the internal error so that way I could differentiate between the different `E_INVALID_REQUEST`).  Similarly, below the error could be `code_verifier failed comparison with code_challenge`.